src/backer/ogmiosing.py

```diff
@@ -0,0 +1,72 @@
+# -*- encoding: utf-8 -*-
+"""
+CARDANO-BACKER
+backer.ogmiosing module
+
+class to support subcribe tip from ogmios and cardano node
+"""
+import ogmios
+from hio.base import doing
+from keri import help
+from ogmios.statequery import QueryNetworkTip
+
+logger = help.ogler.getLogger()
+
+
+class Ogmioser(doing.DoDoer):
+
+    def __init__(self, on_tip=False, tip_slot=0, **kwa):
+        self.on_tip = on_tip
+        self.tip_slot = tip_slot
+        doers = [doing.doify(self.queryTipDo), doing.doify(self.crawlBlockDo)]
+        super(Ogmioser, self).__init__(doers=doers, **kwa)
+
+    def crawlBlockDo(self, tymth=None, tock=0.0):
+        self.wind(tymth)
+        self.tock = tock
+        _ = (yield self.tock)
+
+        try:
+            with ogmios.Client() as client:
+                _, tip, _ = client.find_intersection.execute([ogmios.Origin()])
+                _, _, _ = client.find_intersection.execute([tip.to_point()])
+
+                while True:
+                    direction, tip, point, _ = client.next_block.execute()
+                    if direction and direction == ogmios.Direction.forward:

```

Comment:
Do you know under what circumstances the direction isn't forward?

Reply:
For next_block we have 2 type of direction: forward and backward: ([The reference link here](https://ogmios.dev/mini-protocols/local-chain-sync/))
Ogmios will send back a response which can instruct you either to roll forward or to roll backward. Rolling forward is pretty straightforward and is the main type of response one can expect; such response will include the next block
Rolling backward however may occur when, since the last request, the underlying node decided to switch to a different fork of the chain to the extent that the previous cursor is no longer pointing to a block that exists on the chain

Reply:
Ah yes, we should roll backward too so. Otherwise we will get stuck on the wrong fork. Can you add that to the PR? The tip should decrease

Reply:
Yes for backward, But I am confusing: For each block we crawled, what thing we can process for it? In my current code, I just use to compare to determine the tip.
And for the tip, I have just figured out that we can get it right in the next_block res [(reference here](https://gitlab.com/viperscience/ogmios-python/-/blob/main/examples/find_wallet_transactions.py))
So I will remove queryTipDo (We can add it later if needed)

Reply:
@Sotatek-Patrick-Vu The cardano-backer will receive KELs from agents (like from KERIA) and write KERLs on-chain using ogmios. We need to make sure we are up to tip before we do that.

For now the easiest thing is to not start the backer until we are up to tip. Although we could actually accept KELs into our DB in the background and then write them on-chain once we are the tip - that would be better, maybe in another PR.

Once we've submitted transactions, we will need to make sure they appear on-chain, and if they don't (in case of forks etc), we will need to re-submit them. So we will also check for this in blocks we are crawling in another PR.

Reply:
And great regarding the tip in `next_block` - perfect!